src/middlewares/har.ts

```diff
@@ -0,0 +1,32 @@
+import type { Entry } from "har-format";
+import { NextFunction, Request, Response } from "express";
+import storageService from "../services/storageService";
+import { buildHarRequest, buildHarResponse } from "../core/utils/harFormatter";
+
+
+export const HarMiddleware = (req: Request, res: Response, next: NextFunction) => {
+    const originalSend = res.send;
+    
+    const requestStartTime = new Date();
+    const requestStartTimeStamp: string = requestStartTime.toISOString();
+    
+    let responseBody: string;
+
+    res.send = function (body) {
+        responseBody = body;
+        return originalSend.call(this, body);
+    };
+
+    res.once('finish', () => {
+        const HarEntry: Partial<Entry> = {
+            time: Date.now() - requestStartTime.getTime(),
+            startedDateTime: requestStartTimeStamp,
+            request: buildHarRequest(req),
+            response: buildHarResponse(res, { body: responseBody }),
+        }
+
+        storageService.storeLog({ mockId: res.locals.metadata.mockId, HarEntry, })
```

Comment:
Faced a crash here while running locally
```
/Users/sahilgupta/Documents/dev/requestly/requestly-mock-server/src/middlewares/har.ts:28
        storageService.storeLog({ mockId: res.locals.metadata.mockId, HarEntry, })
                                                              ^
TypeError: Cannot read properties of undefined (reading 'mockId')
```

Reply:
I faced the same error once. Probably because of the order in which we are applying the metadata in `core/server.ts` (_but still the error shouldn't occur_).

It suprisingly went away when I restarted the server and tested with [the backend](https://github.com/requestly/requestly-cloud/pull/399)